src/components/password-strength-field/password-strength.ts

```diff
@@ -0,0 +1,142 @@
+import { Component, Input, SimpleChanges } from '@angular/core';
+import { PasswordService } from '../../services/password-service/password-service';
+import { BulkTranslateService } from '../../services/bulk-translate-service/bulk-translate-service';
+import { TranslateService } from '@ngx-translate/core';
+
+/**
+ * Generated class for the PasswordStrengthComponent component.
+ *
+ * See https://angular.io/api/core/Component for more info on Angular
+ * Components.
+ */
+@Component({
+  selector: 'smilo-password-strength-field',
+  templateUrl: 'password-strength.html'
+})
+export class PasswordStrengthComponent {
+  // The password input to check
+  @Input() password: string;
+  // List of translations set programmatically
+  translations: Map<string, string> = new Map<string, string>();
+  // The password strength object
+  passwordStrength: any;
+  // The password strength between and including 0-4. Default 0
+  passwordStrengthScore: number = 0;
+  // The default password strength score text
+  passwordStrengthScoreText: string;
+  // The array of feedback messages
+  passwordFeedbackWarning: string[] = [];
+  // The online throttling time
+  onlineThrottlingMessage: string = "-";
+  // The online no throttling time
+  onlineNoThrottlingMessage: string = "-";
+  // The offline slow hashing time
+  offlineSlowHashingMessage: string = "-";
+  // The offline fast hashign time
+  offlineFastHashingMessage: string = "-";
+  // The toggle for the advanced content to show
+  showPasswordAdvanced: boolean = false;
+
+  /**
+   * @param passwordService The password service to get the strength from
+   */
+  constructor(private passwordService: PasswordService,
+              private bulkTranslateService: BulkTranslateService,
+              private translateService: TranslateService) { 
+    this.getAndSubscribeToTranslations();
+  }
+
+  /**
+   * Retrieves the translation and again if the language changed
+   */
+  getAndSubscribeToTranslations(): void {
+    this.translateService.onLangChange.subscribe(data => {
+      this.retrieveTranslations();
+    });
+    this.retrieveTranslations();
+  }
+
+  /**
+     * Gets the translations to set programmatically
+     */
+    retrieveTranslations(): Promise<Map<string, string>> {
+      return this.bulkTranslateService.getTranslations([
+          "password-strength-component.fill-strong-password",
+          "password-strength-component.password-weak",
+          "password-strength-component.password-normal",
+          "password-strength-component.password-good",
+          "password-strength-component.password-excellent",
+      ]).then(data => {
+          this.translations = data;
+          this.passwordStrengthScoreText = this.translations.get("password-strength-component.fill-strong-password");
```

Comment:
Nitpicky detail but the retrieveTranslations function also set the passwordStrengthScoreText. Is this really the responsibility of this function or should the calling function handle this detail?